clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/AppengineSafeRetry.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.appengine.deploy;
+
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.clouddriver.appengine.deploy.exception.AppengineOperationException;
+import com.netflix.spinnaker.clouddriver.data.task.Task;
+import com.netflix.spinnaker.clouddriver.googlecommon.deploy.GoogleCommonSafeRetry;
+import com.netflix.spinnaker.clouddriver.googlecommon.deploy.GoogleCommonSafeRetry.GoogleApiException;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
+import groovy.lang.Closure;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNullableByDefault;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+@NonnullByDefault
+public class AppengineSafeRetry {
```

Comment:
Just curious how you decided which refactored classes should become `final` -- is there any reason this one did not?

Reply:
Good catch, I should make this `final` too.